modules/virt-cnv-storage-alerts.adoc

```diff
@@ -0,0 +1,377 @@
+// Module included in the following assemblies:
+//
+// * virt/logging_events_monitoring/virt-events.html/virt-virtualization-alerts.adoc
+
+[id="virt-cnv-storage-alerts_{context}"]
+= Storage Alerts
+
+There are several alerts for CNV Containerizes Data Importer (CDI) operator and Host Path Provisioner (HPP) operator.
+
+//CDIDataImportCronOutdated Alert
+== CDIDataImportCronOutdated alert
+
+.Meaning
+
+DataImportCron polls the latest versions of disk images such as PVCs, commonly into a golden image namespace.
+
+Such PVCs always get updated with the latest version, serving as a trustworthy clone source for VM created from the latest image of an OS.
+
+This alert occurs when a DataImportCron fails to create a corresponding PVC or keep its corresponding PVC updated because it already has `latest` version.
+
+[NOTE]
+====
+If you use a golden image, `latest` is the latest OS version of the distribution.
+If you do not use a non-golden image, `latest` is the latest hash of the image that is available.
+====
+
+.Impact
+
+VMs are created from outdated disk images or VMs fail to start because there is no source PVC from which to clone.
+
+.Diagnosis
+
+*Distinguish between golden images and regular crons*
+
+DataImportCrons are widely used for golden images and likely serve as a source to create VM disks.
+
+* Find the erronous DataImportCron’s namespace & name
++
+[source,terminal]
+----
+oc get dataimportcron -A -o json | jq -r '.items[] | select(.status.conditions[] | select(.type == "UpToDate" and .status == "False")) | .metadata.namespace + "/" + .metadata.name'
+----
+
+Output is returned as `namespace/name - golden image crons` residing in the `openshift-virtualization-os-images` namespace.
+
+* With golden image crons, verify a default storge class is configured.
++
+[source,terminal]
+----
+oc get sc
+----
+
+A single storage class should be marked with (default) next to its name.
+
+*Diagnosis artifacts*
+
+* Substitute `<cron_namespace>` and `<cron_name>` to find the DataImportCron’s corresponding DataVolume that resides in same namespace as cron.
++
+[source,terminal]
+----
+oc -n <cron_namespace> get dataimportcron <cron_name> -o json | jq .status.lastImportedPVC.name
+----
+
+* Substitute `<dv_name>` with the previous output to check for error messages using.
++
+[source,terminal]
+----
+oc -n <cron_namespace> get dv <dv_name> -o yaml
+----
+
+* Find cdi-operator’s pod namespace
++
+[source,terminal]
+----
+export openshift-cnv="$(oc get deployment -A | grep cdi-operator | awk '{print $1}')"
+----
+
+* Check cdi controller logs for error messages.
++
+[source,terminal]
+----
+oc logs -n $openshift-cnv deployment/cdi-deployment
```

Comment:
@mgarrellRH , Lets revert this and keep the $NAMESPACE instead of $openshift-cnv

Reply:
@sradco At this late date, I recommend that you stay with the default "openshift-cnv" instead of a "$NAMESPACE" variable. It is safer, as you say above. :-)

Reply:
Based on chat with Shirly, keeping $NAMESPACE.